src/libraries/Microsoft.Extensions.Configuration.CommandLine/src/CommandLineConfigurationProvider.cs

```diff
@@ -39,96 +39,158 @@ public CommandLineConfigurationProvider(IEnumerable<string> args, IDictionary<st
         public override void Load()
         {
             var data = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
-            string key, value;
 
             using (IEnumerator<string> enumerator = Args.GetEnumerator())
             {
+                bool isNext = enumerator.MoveNext();
+
+                // Store 1st argument here and start while loop with the 2nd,
+                // or 'Current', argument. This is so we can look at the
+                // 'Current' argument in relation to the 1st while evaluating
+                // the 1st.
+                string previousArg = enumerator.Current;
+
+                // If no first arg, return empty dictionary
+                if (!isNext || previousArg == null)
+                {
+                    Data = data;
+                    return;
+                }
+
                 while (enumerator.MoveNext())
                 {
+                    // 'Current' is now the 2nd argument in Args
                     string currentArg = enumerator.Current;
-                    int keyStartIndex = 0;
-
-                    if (currentArg.StartsWith("--"))
+                    KeyValuePair<string, string>? kvp = ProcessArgs(previousArg, currentArg);
+                    // ProcessArgs will return null if previousArg is invalid
+                    if (kvp.HasValue)
```

Comment:
a nit: this can be made a little bit easier to follow (with slightly improved codegen) using `TryX` pattern and tuple return:

```c#
// TryProcessArgs will return false if previousArg is invalid
if (TryProcessArgs(previousArg, currentArg, out (string key, string value) pair))
{
    data[pair.Key] = pair.Value;
}
```

Reply:
Ooo I hadn't thought of that. I like it! Lemme give it a whirl. (Always wanted to implement the TryX pattern, seriously.)

Reply:
Pushed.